modular_skyrat/modules/morenarcotics/code/twitch.dm

```diff
@@ -0,0 +1,181 @@
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = our_guy.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+
+	var/list/col_filter_green = list(0.5,0,0,0, 0,1,0,0, 0,0,0.5,0, 0,0,0,1)
+
+	game_plane_master_controller.add_filter("twitch_filter", 10, color_matrix_filter(col_filter_green, FILTER_COLOR_RGB))
+
+	game_plane_master_controller.add_filter("twitch_blur", 1, list("type" = "radial_blur", "size" = 0.1))
+
+	for(var/filter in game_plane_master_controller.get_filters("twitch_blur"))
+		animate(filter, loop = -1, size = 0.2, time = 2 SECONDS, easing = ELASTIC_EASING|EASE_OUT, flags = ANIMATION_PARALLEL)
+		animate(size = 0.1, time = 6 SECONDS, easing = CIRCULAR_EASING|EASE_IN)
+
+/datum/reagent/drug/twitch/on_mob_end_metabolize(mob/living/carbon/our_guy)
+	. = ..()
+
+	our_guy.remove_movespeed_modifier(/datum/movespeed_modifier/reagent/twitch)
+	our_guy.next_move_modifier += (overdosed ? 0.5 : 0.3)
+
+	our_guy.sound_environment_override = NONE
+
+	speech_effect_span = "hierophant"
+
+	UnregisterSignal(our_guy, COMSIG_MOVABLE_MOVED)
+	UnregisterSignal(our_guy, COMSIG_MOVABLE_HEAR)
+
+	if(constant_dose_time < 100) // Anything less than this and you'll come out fiiiine, aside from a big hit of stamina damage
+		our_guy.visible_message(
+			span_danger("[our_guy] suddenly slows from their inhuman speeds, coming back with a wicked nosebleed!"),
+			span_danger("You suddenly slow back to normal, a stream of blood gushing from your nose!")
+		)
+		our_guy.adjustStaminaLoss(constant_dose_time)
+	else // Much longer than that however, and you're not gonna have a good day
+		our_guy.visible_message(
+			span_danger("[our_guy] suddenly snaps back from their inhumans speeds, coughing up a spray of blood!"),
+			span_danger("As you snap back to normal speed you cough up a worrying amount of blood. You feel like you've just been run over by a power loader.")
+		)
+		our_guy.spray_blood(our_guy.dir, 2) // The before mentioned coughing up blood
+		our_guy.emote("cough")
+		our_guy.adjustStaminaLoss(constant_dose_time)
+		our_guy.adjustOrganLoss(ORGAN_SLOT_HEART, 0.3 * constant_dose_time) // Basically you might die
+
+	if(!our_guy.hud_used)
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = our_guy.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+
+	game_plane_master_controller.remove_filter("twitch_filter")
+	game_plane_master_controller.remove_filter("twitch_blur")
+
+/// Leaves an afterimage behind the mob when they move
+/datum/reagent/drug/twitch/proc/on_movement(mob/living/carbon/our_guy)
+	SIGNAL_HANDLER
+	new /obj/effect/temp_visual/decoy/twitch_afterimage(our_guy.loc, our_guy)
+
+/datum/reagent/drug/twitch/on_mob_life(mob/living/carbon/our_guy, delta_time, times_fired)
+	. = ..()
+
+	constant_dose_time += delta_time
+
+	for(var/possible_kronk in our_guy.reagents.reagent_list) // Kronkaine, another heart-straining drug, could cause problems if mixed with this
```

Comment:
couldn't 121-122 be `if(locate(/datum/reagent/drug/kronkaine) in our_guy.reagents.reagent_list)`?

Reply:
Probably could be